tests/test_pyright.py

```diff
@@ -0,0 +1,22 @@
+import pytest
+from pathlib import Path
+from shutil import which
+from subprocess import run
+
+topdir = Path(__file__).resolve().parent.parent
+
+pyright = which("pyright")
```

Comment:
How about adding this as a dev dependency in `setup.py` (as we do for `mypy` and `flake8`), which would then make `pyright` available to all developers, and avoid the if/else + test skipping added here. We'd also avoid using `npm/npx` within augur, which feels strange to me. (This also applies to the changes in `DEV_DOCS.md`.)

Reply:
+1 to not crossing the streams

Reply:
The `pyright` distribution on PyPI is unofficial and all it does is set up Node and invoke npm behind the scenes to install the official NPM `pyright` distribution. It's still "crossing the streams", just behind your back.

It seems more straightforward and less prone to issues with Node envs to do this ourselves instead. I personally don't really see an issue with mixing package ecosystems/managers when they're disjoint.

(Background: Victor got this approach from my setup for Nextstrain CLI. The `pyright` package on PyPI was introduced right around the same time I started using Pyright in the CLI, and I'm not sure if I was aware of then or not.)

Reply:
Continuing in #1468